app/controllers/api_controller/providers.rb

```diff
@@ -79,6 +85,9 @@ def create_provider(data)
     end
 
     def edit_provider(provider, data)
+      check_connection_configurations(data)
+      data["connection_configurations_array"] =
+        merge_connection_configurations(provider, data) if data.key?("connection_configurations_array")
```

Comment:
The current interface of `merge_connection_configurations` means both it and the caller know about the `"connection_configurations_array"` key.
- The caller here checks the key's presence and stores the result.
- The function receives whole data, but fetches `data[:connection_configurations_array]` (via `data_connection_configurations(data)`).
BTW, is it OK that this key is sometimes string  sometimes symbol?